webClient/src/app/shared/editor-control/editor-control.service.ts

```diff
@@ -583,6 +585,130 @@ export class EditorControlService implements ZLUX.IEditor, ZLUX.IEditorMultiBuff
+    //TODO: validation of record length that is aware of how DBCS will effect actual length
+    //FB must have exactly lrecl, VB must have no more than lrecl. content cant be undefined.
+    if (!content) {
+      return 'No Content';
+    }
+    if(datasetAttrs.recfm.recordLength === 'U') {
+      return 'Cannot save a dataset with Undefined record format';
+    }
+    let maxRecordLen = datasetAttrs.dsorg.maxRecordLen;
+    for (let i = 0; i < content.length; i++) {
+      let record = content[i];
+      if (record.length > maxRecordLen) {
+        return `Line ${i+1} exceeds record length limit of ${maxRecordLen}`;
+      }
+    }
+    if (JSON.stringify({records:content}).length > MAX_CONTENT_LENGTH) {
+      return 'Content over max size currently supported ('+ (MAX_CONTENT_LENGTH/(1024*1024)) +'MB)';
+    }
+    return true;
+  }
+  
+  saveDatasetHandler(context?: ProjectContext, destinationOverride?: any): Observable<void> {
+    const _openDataset = this.openFileList.getValue();
+    const editor = this._editor.getValue();
+    const forceWrite = false;
+    let _activeDataset: ProjectContext;
+    let _observer: Observer<void>;
+    let _observable: Observable<void>;
+    let contents;
+    if (editor) {
+      contents = editor.getValue();
+    }
+    if (context != null) {
+      _activeDataset = context;
+    } else {
+      _activeDataset = _openDataset.filter(dataset => dataset.active === true)[0];
+    }
+    const model = _activeDataset.model;
+    const fullName = _activeDataset.model.fileName;
+
+    _observable = new Observable((observer) => {
+      _observer = observer;
+    });
+
+    if (contents === undefined) {
+      this.snackBar.open(`Cannot retrieve contents to save`, "Close", {duration: MessageDuration.Medium, panelClass: 'center'});
+      return;
+    }
+
+    if (!destinationOverride && fullName) {
+      contents = editor.getValue().split('\n');
+      const requestUrl = ZoweZLUX.uriBroker.datasetContentsUri(fullName);
+      this.log.debug(`Should save contents to dataset. dataset=${fullName}, route=${requestUrl}`);
+      let result = this.isContentValidForDatasetWrite(contents, model.datasetAttrs);
+      if (result === true) {
+        this.saveDataset(context, _activeDataset, forceWrite, _observer, _observable);
+      } else {
+        this.snackBar.open(`Content invalid for saving to dataset. Reason=${result}`, 'Close', { duration:MessageDuration.Long, panelClass: 'center'});
+      }
+    } else {
+      //dataset is new or needs some alteration we can't do yet
+      this.snackBar.open(`${_activeDataset.name} could not be saved, feature not yet implemented.`, 
+                         'Close', { duration: MessageDuration.Long,   panelClass: 'center' });    
+    }
+    return _observable;
+  }
+
+  saveDataset(context: ProjectContext, activeDataset: ProjectContext, forceWrite: Boolean, _observer: Observer<void>, _observable: Observable<void>) {
```

Comment:
You set forceWrite to be 'false' (boolean) but made it an Object Boolean, so it just made its truthiness set to false, which could cause bugs